File: snies-backend/audit/domain/entities/audit_log.py
from dataclasses import dataclass
from datetime import datetime


@dataclass(frozen=True)
class AuditLog:
    id: int | None
    created_at: datetime | None
    action: str
    method: str
    path: str
    status_code: int
    user_id: int | None = None
    user_email: str | None = None
    user_role: str | None = None
    ip: str | None = None
    user_agent: str | None = None
    view_name: str | None = None
    module: str | None = None
    resource_id: str | None = None
    request_data: dict | None = None
    response_data: dict | None = None

File: snies-frontend/modules/audit/types/audit-log.ts
export type AuditAction = "create" | "update" | "delete" | string;

export type AuditLog = {
  id: number;
  created_at: string;
  action: AuditAction;
  method: string;
  path: string;
  status_code: number;
  user_id: number | null;
  user_email: string | null;
  user_role: string | null;
  ip: string | null;
  user_agent: string | null;
  view_name: string | null;
  module: string | null;
  resource_id: string | null;
  request_data: unknown;
  response_data: unknown;
};

export type PaginatedAuditLogs = {
  count: number;
  page: number;
  page_size: number;
  results: AuditLog[];
};

